src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentSearchResourceIT.java

```diff
@@ -364,21 +364,15 @@ public void shouldReturn6Records_whenSearchingByFailedExternalState() {
                 .withMandateFixture(mandateFixture)
                 .withState(PaymentState.CREATED)
                 .insert(testContext.getJdbi());
-        for (int i = 0; i < 6; i++) {
-            if (i % 2 == 0) {
-                aPaymentFixture()
-                        .withMandateFixture(mandateFixture)
-                        .withState(PaymentState.CANCELLED)
-                        .withStateDetails("state_details")
-                        .insert(testContext.getJdbi());
-                continue;
-            }
+        
+        List.of(1,2,3).forEach(i -> {
             aPaymentFixture()
                     .withMandateFixture(mandateFixture)
-                    .withState(PaymentState.EXPIRED)
+                    .withState(PaymentState.CANCELLED)
```

Comment:
I guess before it was creating a few paymentFixtures with two internal states what mapped to external failed. Since the mapping isn't really being tested here it seems reasonable to set them all up with the same. Alternatively you could use the `List.of().stream` method you have and put the various states in there `List.of(CANCELLED, FAILED, CUSTOMER_APPROVED_DENIED).stream().forEach(state -> .....)`